Apps/Src/ReadSwitches.c

```diff
@@ -0,0 +1,225 @@
+                            (OS_OPT) OS_OPT_POST_ALL,
+                            (OS_ERR*) &err
+                        );
+		            }
+                    if(motorConThreadSpawned==false){
+                        OSTaskCreate(
+                            (OS_TCB*)&MotorConnection_TCB,
+                            (CPU_CHAR*)"Motor Connection",
+                            (OS_TASK_PTR)Task_MotorConnection,
+                            (void*)NULL,
+                            (OS_PRIO)TASK_MOTOR_CONNECTION_PRIO,
+                            (CPU_STK*)ArrayConnection_Stk,
+                            (CPU_STK_SIZE)sizeof(MotorConnection_Stk)/10,
+                            (CPU_STK_SIZE)sizeof(MotorConnection_Stk),
+                            (OS_MSG_QTY)NULL,
+                            (OS_TICK)NULL,
+                            (void*)NULL,
+                            (OS_OPT)(OS_OPT_TASK_STK_CLR),
+                            (OS_ERR*)&err
+                        );
+                        motorConThreadSpawned=true;
+                    }else {
+                        OSSemPost(
+                            &MotorConnectionChange_Sem4,
+                            (OS_OPT) OS_OPT_POST_ALL,
+                            (OS_ERR*) &err
+                        );
+                    }
+                case 0x02:
+                    return -1; //Shouldn't happen - error case; only IGN2 is on
+            }
+       
+        }
+        
+        //Signal LightsChange_Sem4 for UpdateLights Thread
+        State curLightSwitches[]={
+            currSwitchStates.lightSwitches.LEFT_SW,
+            currSwitchStates.lightSwitches.RIGHT_SW,
+            currSwitchStates.lightSwitches.HEADLIGHT_SW,
+            currSwitchStates.lightSwitches.HZD_SW
+        }; 
+
+        State prevLightSwitches[]={
+            globalSwitchStates->lightSwitches.LEFT_SW,
+            globalSwitchStates->lightSwitches.RIGHT_SW,
+            globalSwitchStates->lightSwitches.HEADLIGHT_SW,
+            globalSwitchStates->lightSwitches.HZD_SW
+        };       
+        uint8_t lightSwitchStates = 0x00; //holds states read from Switches_Driver
+        uint8_t lightSwitchStored = 0x00; //holds states stored in global
+        for(uint8_t i = 0; i<4; i++){
+            lightSwitchStates |= (curLightSwitches[i]<<i);
+            lightSwitchStored |= (prevLightSwitches[i]<<i);
+        }
+        
+        if(lightSwitchStored!=lightSwitchStates){
+            //if any of the bits dont match, CarState needs to get updated and LightsChange must get signaled
+            //TODO: use memcpy to move currSwitchStates.lightSwitches into globalSwitchStates->lightSwitches
+            memcpy(&(globalSwitchStates->lightSwitches),&(currSwitchStates.lightSwitches),sizeof(currSwitchStates.lightSwitches));
+            OSSemPost(
+                &LightsChange_Sem4,
+                (OS_OPT) OS_OPT_POST_ALL,
+                (OS_ERR*)&err
+            );
```

Comment:
Why would you not just make a function to set the states of the different lights and call that? I don't see why you need to populate a global data structure and post a semaphore to make a task run to change the lights.